test/built-ins/RegExp/prototype/Symbol.search/success-get-index-err.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 21.2.5.9
+description: >
+    Behavior when error thrown while accessing `index` property of match result
+info: >
+    [...]
+    14. Return Get(result, "index").
+features: [Symbol.search]
+---*/
+
+var poisonedIndex = {
+  get index() {
+    throw new Test262Error();
+  }
+};
+var fakeRe = {
+  exec: function() {
```

Comment:
@jugglinmike, what do you think about adding new tests to verify that the result object returned from `exec` is either `Object` or `Null`?

Step `5.d.` in Section `21.2.5.2.1` (RegExpExec) says:

    If Type(result) is neither Object or Null, throw a TypeError exception.